### PR TITLE
Show source information data action

### PR DESCRIPTION
**Pull request by gregquinn2001**
JIRA 1102. The action in the targeted data panel visualizer menu for show source note now can show 3 different options depending on the item clicked on: (1) show source note (if there is a reference to a source note, (2) source source (if there's some data available on who authored and/or informed the data entry and/or when it was created), and (3) no source information. The first one opens the source note as before. The 2nd one shows a modal with the available source information shown. The final one just shows the snackbar still indicating that no source info.

I added some data to the sarcoma patient for testing and demonstrating so I'd recommend looking at it on there some - pilot1 endpoint.

I also removed the "Current Diagnosis" subsection heading in the Summary section for both cancer condition summaries. It caused confusion during CancerLinQ demo and isn't really accurate. Of course, let me know if you think otherwise.

JIRA 1255. Sarcoma patient (pilot1) most recent disease status is now the one in the 1 note and they are linked.

_Pull requests into Flux require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:._

## Submitter:

**Running the Application**

- [X] Manually tested in Chrome
- [ ] Manually tested in IE

**Documentation**

- [X] This pull request describes why these changes were made
- [ ] Recognizes any potential shortcomings/bugs in the description 
- [ ] Cheat sheet is updated
- [ ] Demo script is updated 
- [ ] Documentation on Wiki has been updated 
- [ ] Additional technical components and libraries are documented and added to wiki as needed

**NoteParser**

- [ ] Note parser has been updated if structured phrases change

**Code Quality**

- [ ] 4-space indents - convert any tabs to spaces
- [ ] Use public class field syntax for binding functions - ex. handleChange = () => { ... };
- [ ] Code is commented

**Tests**

- [ ] Existing tests passed
- [ ] Added Enzyme-UI tests for slim mode 
- [ ] Added Enzyme-UI tests for full mode
- [ ] Added backend tests for new functionality added
- [ ] Added note parser examples to test new functionality

## Reviewer 1: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

## Reviewer 2: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: src/summary/SummaryMetadata.jsx [hunk 1]
```diff
@@ -1346,26 +1349,41 @@ export default class SummaryMetadata {
         return this.hardCodedMetadata;
     }
 
+    determineSource = (patient, entry) => {
+        if (entry.sourceClinicalNoteReference) return entry.sourceClinicalNoteReference;
+        let result = "";
+        if (entry.author) result += "recorded by " + entry.author;
```
Comment (mgramigna) [inline src/summary/SummaryMetadata.jsx]: [MG1] Very small issue, but recorded should probably be capitalized here.

**Comment by gregquinn2001** [comment]
MG1 addressed

**Comment by gregquinn2001** [comment]
MG2: I agree but never followed through on that change. I'll take a quick look and see if I can disable the menu item.

**Comment by gregquinn2001** [comment]
MG2 implemented. No source information is now disabled.

**Comment by gregquinn2001** [comment]
All issues addressed!

**Comment by danlee1025** [comment]
This looks very cool!  All backend tests pass, just needs to be rebased and I will approve.

**Pull request merged**